src/libostree/ostree-sysroot.c

```diff
@@ -813,6 +815,17 @@ ostree_sysroot_load_if_changed (OstreeSysroot  *self,
                                cancellable, error))
     return FALSE;
 
+  /* Determine whether we're "physical" or not, the first time we initialize */
+  if (!self->loaded)
+    {
+      if (fstatat (self->sysroot_fd, "sysroot", &stbuf, 0) < 0)
```

Comment:
Hmm, this still feels a bit weak/arbitrary. Wouldn't it be safer to literally check if `self->path` corresponds to one of the picked up deployment dirs? E.g. by just cross-checking inodes? Or do we want to support initializing an `OstreeSysroot` to a deployment dir that's on a separate physical storage than `/`? (But then it'd seem more sane to create an `OstreeSysroot` to that physical sysroot storage instead anyway).

What I mean is, instead of checking that it's physical, it seems like it would be safer to check that it's *not* physical, since we're in pretty tight control of deployment dir layouts, etc..., right? (Whereas OTOH it seems like we shouldn't be affected by whatever else other than `/ostree` might be on the physical sysroot).